### ISSUE TITLE
Individual Key Bindings

### ISSUE THREAD

**Issue by trisys3** [jorgebucaran/fisher#534]
It seems like individual key bindings that we had before installing `fisher` are being stripped out. I previously had "ctrl-g" mapped to the previous behavior of "ctrl-c", to remove the current line without making a new one. I put this in my `fish_user_key_bindings` function, but this function gets completely rewritten by `fisher`, and it seems the `fisher` bindings clobbered mine. I tried to add the binding to my `fish_config` file, to no avail, and I'm afraid if I add it to the key bindings function again it'll get overridden on update.

My question is twofold:

* Is there a way to add custom key bindings, and
* Is there a way to prevent `fisher` from clobbering previously-added key bindings when installing or updating?

The binding I added was `bind \cg 'commandline ""'`, added to `~/.config/fish/functions/fish_user_key_bindings.fish`.

For completeness, I have `fish` version 2.7.1-1429-gdaf520d, installed with `apt-get`. Obviously, I have an `ubuntu`, version 19.04 I believe. Not sure if this matters, but the computer is from System76, meaning Ubuntu is installed directly on the laptop.  Thus, even though I have a much newer version, when I do e.g. `lsb_release -a`, it still shows 16.04 "Xenial".

**Comment by jorgebucaran** [comment]
@trisys3 Can you provide a way to reproduce the problem? Fisher is not clobbering previously-added key bindings here and I can successfully use multiple packages that install custom key bindings.

**Comment by jorgebucaran** [comment]
@trisys3 Ping

**Comment by trisys3** [comment]
I had a `fish_user_key_bindings.fish` file in my `~/.config` folder, and I installed `fisher`. I also installed `fish-nvm`, so I'm not sure exactly which one removed my previous binding. I will try to experiment when I get the chance. This happened twice, on two different computers, so I know it's not a fluke.

By the way, just to be completely clear here, I'm not adding a package that has some key bindings attached. I am adding my own, custom key binding, and expecting it to stick around when I install `fisher`.

**Comment by jorgebucaran** [comment]
Gotcha. @trisys3 I use custom key bindings myself too. I put them in `~/.config/fish/functions/fish_user_key_bindings.fish` and installing packages doesn't break them. Have a look also at https://github.com/jethrokuan/fzf/pull/85 for more information.

If you find a way to reproduce the problem (I really hope you do!), please reopen and explain how; for example, using Docker, running a number of commands in a particular way, and so on.

**Comment by trisys3** [comment]
I will try to replicate the problem this week or maybe over the weekend.

**Comment by trisys3** [comment]
Hmmm... I just uninstalled & re-installed `fisher` and `fish-nvm` and didn't confront the same issue. I had had an older version of `fisher`, as well as `fnm`, the previous incarnation of `fish-nvm`, so maybe that was what happened. Alternatively, I had also installed `fisherman`, which I believe is the previous incarnation of `fisher`? Regardless, thanks for being so patient and sorry for the noise. If I do see this again, I will open another issue.
